src/Stryker.Core/Stryker.Core/MutantFilters/DiffMutantFilter.cs

```diff
@@ -1,33 +1,218 @@
-﻿using Stryker.Core.DiffProviders;
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.Text;
+using Microsoft.Extensions.Logging;
+using Stryker.Core.Clients;
+using Stryker.Core.DashboardCompare;
+using Stryker.Core.DiffProviders;
+using Stryker.Core.Logging;
 using Stryker.Core.Mutants;
 using Stryker.Core.Options;
 using Stryker.Core.ProjectComponents;
+using Stryker.Core.Reporters.Json;
+using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Threading.Tasks;
 
 namespace Stryker.Core.MutantFilters
 {
     public class DiffMutantFilter : IMutantFilter
     {
         private readonly DiffResult _diffResult;
+        private readonly IDashboardClient _dashboardClient;
+        private readonly IBranchProvider _branchProvider;
+
+        private readonly StrykerOptions _options;
+
+        private readonly JsonReport _baseline;
+
+        private readonly ILogger<DiffMutantFilter> _logger;
+
         public string DisplayName => "git diff file filter";
 
-        public DiffMutantFilter(IDiffProvider diffProvider)
+        public DiffMutantFilter(StrykerOptions options = null, IDiffProvider diffProvider = null, IDashboardClient dashboardClient = null, IBranchProvider branchProvider = null)
         {
-                _diffResult = diffProvider?.ScanDiff();
+            _logger = ApplicationLogging.LoggerFactory.CreateLogger<DiffMutantFilter>();
+
+            _dashboardClient = dashboardClient ?? new DashboardClient(options);
+            _branchProvider = branchProvider ?? new GitBranchProvider(options);
+            _options = options;
+
+            if (options.CompareToDashboard)
+            {
+                _baseline = GetBaseline().Result;
+            }
+
+            _diffResult = diffProvider.ScanDiff();
         }
 
         public IEnumerable<Mutant> FilterMutants(IEnumerable<Mutant> mutants, FileLeaf file, StrykerOptions options)
         {
+            if (options.CompareToDashboard)
+            {
+                if (_baseline == null)
+                {
+                    return mutants;
+                }
+
+                UpdateMutantsWithBaseline(mutants, file);
+            }
+
             if (!_diffResult.TestsChanged)
```

Comment:
Could you add comments on each decision in this tree for why the decision is made and what it means for the result?

Reply:
So for SetMutantStatusForFileChanged, ResetStatusForStatusUnclear and ResetStatusForStatusUnclear